PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/WhatsNewHelper.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.helpers;
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.lang.Language;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.testFramework.LightVirtualFile;
+import com.microsoft.azure.hdinsight.common.StreamUtil;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.intellij.plugins.markdown.ui.preview.MarkdownSplitEditor;
+import org.intellij.plugins.markdown.ui.split.SplitFileEditor;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public enum WhatsNewHelper {
+    INSTANCE;
+
+    private static final String AZURE_TOOLKIT_FOR_JAVA = "Azure Toolkit for Java";
+    private static final String AZURE_TOOLKIT_WHATS_NEW = "AzureToolkit.WhatsNew";
+    private static final String VERSION_PATTERN = "<!-- Version: (.*) -->";
+    private static final String WHAT_S_NEW_CONSTANT = "WHAT_S_NEW";
+    private static final String WHAT_S_NEW_CONTENT_PATH = "/whatsnew/whatsnew.md";
+    private static final Key<String> WHAT_S_NEW_ID = new Key<>(WHAT_S_NEW_CONSTANT);
+
+    public synchronized void showWhatsNew(boolean force, Project project) throws IOException {
```

Comment:
I notice there are `NotNull` and `Nullable` annotations in the code above, how about also apply to this file? Btw, in my practise, we can implicit the type is `NotNull` which means only have to annotate `Nullable`.

Reply:
Added the `NotNull` mark to public method